packages/deployer/src/utils/validation.ts

```diff
@@ -0,0 +1,13 @@
+import { Config } from '@api3/node';
+import * as logger from '../utils/logger';
+
+export function validateConfig(config: Config, nodeVersion: string) {
+  if (nodeVersion !== config.nodeSettings.nodeVersion) {
+    logger.fail(
+      `nodeVersion under nodeSettings in config.json is ${config.nodeSettings.nodeVersion} while the deployer node version is ${nodeVersion}`
+    );
+    throw new Error('Attempted to deploy node configuration with the wrong node version');
```

Comment:
This message is a bit misleading/confusing to me. I had to read through the code to actually understand what it means. I would rather phrase it as something like this. What do you think?

```suggestion
    throw new Error('Node version specified in config.json does not match the deployer's version');
```